lib/private/Accounts/AccountManager.php

```diff
@@ -144,6 +144,17 @@ public function updateUser(IUser $user, array $data, bool $throwOnData = false):
 			}
 		}
 
+		// set a max length
+		foreach ($data as $propertyName => $propertyData) {
+			if (isset($data[$propertyName]) && isset($data[$propertyName]['value']) && strlen($data[$propertyName]['value']) > 2048) {
+				if ($throwOnData) {
+					throw new \InvalidArgumentException($propertyName . ' is too long');
```

Comment:
```suggestion
					throw new \InvalidArgumentException($propertyName);
```

This way the clients receive the field ID and can show a proper localized message